### ISSUE TITLE
/oauth2/token endpoint is being deprecated 

### ISSUE THREAD

**Issue by virgytam** [stevenmaguire/yelp-php#23]
Hi Steven, Virginia from Yelp here! 👋

Just a friendly reminder that the oauth2/token endpoint is being deprecated on March 1, 2018. After this date, any calls to this endpoint will result in an error. Access Tokens are being deprecated in favor of API Keys that have no expiration date and are used exactly like access tokens. (for more info, see https://www.yelp.com/developers/documentation/v3/authentication#where-is-my-client-secret-going)

I noticed that this clientlib uses your oauth2-yelp clientlib, which still has a reference to this endpoint and wanted give you a heads up incase any integrations break on the March 1st date.

**Comment by stevenmaguire** [comment]
Thanks for the heads up, @virgytam! I will get an updated version of the package together that supports API key authentication in favor of access tokens. Will the API itself see a major version change here?

**Comment by virgytam** [comment]
No major version change, the API will continue to be v3 and all endpoints will have v3 in the name

**Comment by stevenmaguire** [comment]
@virgytam thanks! After reviewing the change, it appears to be very minor so I've got a fix ready to go: 

https://github.com/stevenmaguire/yelp-php/pull/25

Technically, the library would've still worked with developers providing their API Key as `accessToken`, and I want to keep developer experience pretty high so I added superficial support for `apiKey` as well as `accessToken`, preferring `apiKey` if both provided. The client should work well!

Thanks for the heads-up!
